Sources/Nats/NatsConnection.swift

```diff
@@ -511,8 +499,23 @@ class ConnectionHandler: ChannelInboundHandler {
     func close() async throws {
         self.reconnectTask?.cancel()
         await self.reconnectTask?.value
-        self.state = .closed
-        try await disconnect()
+
+        let eventLoop = self.channel?.eventLoop ?? MultiThreadedEventLoopGroup.currentEventLoop!
+        let promise = eventLoop.makePromise(of: Void.self)
+
+        eventLoop.execute {  // This ensures the code block runs on the event loop
+            Task {
+                self.state = .closed
+                do {
+                    try await self.disconnect()
```

Comment:
disconnect is essentially a call to `channel.close()` and can be done without spawning a `Task`.

Reply:
You're right. I wanted to re-use disconnect method but it does not make sense here.